src/client/Microsoft.Identity.Client/AuthenticationHeaderParser.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parsed authentication headers to retrieve header values from HttpResponseHeaders.
+    /// </summary>
+    public class AuthenticationHeaderParser
```

Comment:
Since the public API is stablized, could you please create a class diagram or a table with all the public API in these classes, ensuring that you mark: 

- what API is old
- what API is new
- what API has been deprecated and what we're suggesting to replace it with

This will make it easier to review this doc and should be added to the API Review spec as well.

Reply:
Updated PR description